scalableminds/webknossos
# Repair Unused DBAccessContexts

There were a lot of unused warnings for DBAccessContext objects, which are used on the lowest level to ensure rights management. 
- Most of these were correctly unused, in this case I removed them entirely. 
- Some were unused correctly but could not be removed because of subclassing, in this case I added `@silent`
- Some pointed to missing access checks, which I added, either in form of DB access checks or explicit checks on higher levels
- I also added access checks to the reports controllers to ensure reports can only be obtained for the own organization and teams (even if team IDs are leaked somehow)
- One remaining problem is with ExperienceDomains, tracked in follow-up issue #5071
- I also included some small refactorings here (sorry for the large diff, but most small changes should be obviously harmless)

### Steps to test:
- upload dataset with initial teams (as admin/dataset manager) should still be allowed
- project publicWrites was changed (hide owner if user is not allowed to see owner), should check if that creates problems (try to trace tasks as normal user)
- update description / name of annotation
- dataset token sharing should still work
- public annotations should still be visible as logged-out user
- my annotations list should still work
- you should no longer get a response for report requests even if you know the right team id, if you are not in that team and no admin

### Issues:
- fixes #5013
- fixes #4992

------
